flamedisx/likelihood.py

```diff
@@ -93,17 +109,14 @@ def __init__(
             sname: sclass.find_defaults()[2]
             for sname, sclass in self.sources.items()}
 
-        # Create sources. Have to copy data, it's modified by Source.set_data
+        # Create sources
         self.sources = {
-            sname: sclass(data=(None
-                                if data[self.d_for_s[sname]] is None
-                                else data[self.d_for_s[sname]].copy()),
+            sname: sclass(data=None,
```

Comment:
This means we don't set the batch_size based on data size any more.
In source when data is not None we do: `self.batch_size = min(batch_size, len(data))`
Not sure if still needed.

Reply:
Good point, added a batch size check in the likelihood init